frontend/src/components/Collections/components/Grid/components/CollectionsGrid/index.tsx

```diff
@@ -80,10 +78,20 @@ function renderCollections(
   return collectionElements;
 }
 
-function sortByCreatedAtDescending(
-  collections: RevisionResponse[]
-): RevisionResponse[] {
-  return collections?.sort((a, b) => b.created_at - a.created_at) || [];
+function sortByCreatedAtDescendingAndFormat(
+  collectionsMap: CollectionResponsesMap
+): CollectionResponse[] {
+  return (
+    Array.from(collectionsMap.values())
+      .map((collectionsWithIDMap) => {
+        return (
+          collectionsWithIDMap.get(VISIBILITY_TYPE.PUBLIC) ||
+          collectionsWithIDMap.get(VISIBILITY_TYPE.PRIVATE) ||
+          ({} as CollectionResponse)
```

Comment:
nit: this doesn't seem like a good idea since the properties for this object are all required. it would probably be better to change the return type to something like `Array<CollectionResponse | null>` or maybe make the return type partial `Partial<CollectionResponse>[]` for null safety.

maybe you could also filter out items that aren't in `collectionsWithIDMap`.

Reply:
The map is guaranteed to have at least one of the two `VISIBILITY_TYPE`'s, the empty object is just for type safety since `.get` can return null.  Is there a better way to do this?

Reply:
> The map is guaranteed to have at least one of the two VISIBILITY_TYPE's

Hmm, what you have is fine then since I didn't know this. Although for readability, I would add a comment or change the code to:

```ts
return (
  collectionsWithIDMap.get(VISIBILITY_TYPE.PUBLIC) ||
  collectionsWithIDMap.get(VISIBILITY_TYPE.PRIVATE)
) as CollectionResponse
```

Reply:
ah that's a great idea.  I'll do that.  Thanks!